pkg/operator/starter.go

```diff
@@ -45,53 +50,90 @@ const (
 	resync = 20 * time.Minute
 )
 
-func RunOperator(ctx context.Context, controllerConfig *controllercmd.ControllerContext) error {
-	kubeClient, err := kubeclient.NewForConfig(rest.AddUserAgent(controllerConfig.KubeConfig, targetName))
+func RunOperator(ctx context.Context, controllerConfig *controllercmd.ControllerContext, guestKubeConfigFile string) error {
+	isHyperShift := guestKubeConfigFile != ""
+	// Kubeconfig received through service account or --kubeconfig arg
+	// is for the management cluster.
+	controlPlaneKubeConfig := controllerConfig.KubeConfig
+	controlPlaneKubeClient, err := kubeclient.NewForConfig(rest.AddUserAgent(controlPlaneKubeConfig, targetName))
 	if err != nil {
 		return err
 	}
-	kubeInformersForNamespaces := v1helpers.NewKubeInformersForNamespaces(kubeClient, "", operatorNamespace, targetNamespace)
+	controlPlaneNamespace := defaultOperandNamespace
+	// Guest kubeconfig is the same as the management cluster one unless guestKubeConfigFile is provided
+	guestKubeClient := controlPlaneKubeClient
+	guestKubeConfig := controlPlaneKubeConfig
+	if guestKubeConfigFile != "" {
+		guestKubeConfig, err = client.GetKubeConfigOrInClusterConfig(guestKubeConfigFile, nil)
+		if err != nil {
+			return fmt.Errorf("Failed to use guest kubeconfig %s: %s", guestKubeConfigFile, err)
+		}
+		controlPlaneNamespace = controllerConfig.OperatorNamespace
```

Comment:
I think this can be simplified by replacing line 62 above with this. The control plane NS should always be the one that this operator is running on. Only the guest NS is different on Hypershift.

Reply:
Nice! Fixed.